simplecpp.cpp

```diff
@@ -779,7 +779,7 @@ static bool isFloatSuffix(const simplecpp::Token *tok)
 {
     if (!tok || tok->str().size() != 1U)
         return false;
-    const char c = std::tolower(tok->str()[0]);
+    const char c = (char)std::tolower(tok->str()[0]);
```

Comment:
I don't have a strong opinion but static_cast is pretty verbose in this case.

If we know that the first char ascii value is 0-127 then the cast is not needed. well I guess for garbage input the value could be out of range so the cast makes sense.

Reply:
I believe I added the casts to please the compiler.  Like you said, your solution is verbose, and I'm not sure what it buys us (perhaps I just don't understand the intricacies of `static_cast<...>`).  Also, since `std::string` is made up of elements of type `char`, it _is_ intended to handle only ASCII values 0-127.  If you want to handle values 0-255, I think it would be better to do something like
```
typedef std::basic_string<unsigned char> TokenString;
```
and then to change `char` to `unsigned char` throughout.

I guess a broader question is whether modern preprocessors should also handle UTF8/Unicode.  I have absolutely no opinion on this.

Reply:
> Also, since std::string is made up of elements of type char, it is intended to handle only ASCII values 0-127.

If utf-8 is used then on my machine since char is signed there will sometimes be negative values. 

I don't know if there is anything unintended about that. I.e. as far as I know there is not undefined behavior. But maybe something in the conversion is undefined.

> typedef std::basic_string<unsigned char> TokenString;

ouch. I believe that would complicate the interface. For instance, Cppcheck reads the tokens from simplecpp and expects that TokenString can be converted directly to plain simple std::string.